skaffold-plugins-core/src/main/java/com/google/cloud/tools/skaffold/downloader/Downloader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.skaffold.downloader;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+/** Downloads files. */
+public class Downloader {
+
+  private final URL url;
+
+  /**
+   * Instantiates with a {@link URL} to download from.
+   *
+   * @param url the {@link URL} to download
+   */
+  public Downloader(URL url) {
```

Comment:
What's the rationale for an instance? Can't it be just `Downloader.download(url, path)`? If for testing purpose in another class, you could still inject `Downloader`, e.g., with a lambda. Even in such a case, it `download()` method could still accept an `URL`. Is it for downloading from the same `URL` source into multiple `Path` locations?